### ISSUE TITLE
Visual C++ constexpr build error for quaternions

### ISSUE THREAD

**Issue by martinfalk** [g-truc/glm#794]
When using using glm::quat I get error C2476 in VS2015 Update 3:
```
ext\glm\glm\glm\gtc\quaternion.inl(90): error C2476: 'constexpr' constructor does not initialize all members
  ext\glm\glm\glm\gtc\../gtc/quaternion.hpp(55): note: 'glm::tquat<float,0>::<unnamed-tag>::x' was not initialized by the constructor
  ext\glm\glm\glm\gtc\../gtc/quaternion.hpp(55): note: 'glm::tquat<float,0>::<unnamed-tag>::y' was not initialized by the constructor
  ext\glm\glm\glm\gtc\../gtc/quaternion.hpp(55): note: 'glm::tquat<float,0>::<unnamed-tag>::z' was not initialized by the constructor
  ext\glm\glm\glm\gtc\../gtc/quaternion.hpp(55): note: 'glm::tquat<float,0>::<unnamed-tag>::w' was not initialized by the constructor
  ext\glm\glm\glm\gtc\../gtc/quaternion.hpp(56): note: 'glm::tquat<float,0>::data' was not initialized by the constructor
  ext\glm\glm\glm\gtc\quaternion.inl(94): note: while compiling class template member function 'glm::tquat<float,0>::tquat(void)'
  src\core\interaction\trackball.cpp(118): note: see reference to function template instantiation 'glm::tquat<float,0>::tquat(void)' being compiled
```

I tracked it down to `quaternion.hpp` and `quaterion.inl` where `GLM_CONSTEXPR` is used for all its constructors. 
Replacing these with `GLM_CONSTEXPR_CTOR` seemed to solve the issue for me.

This was partially fixed for the vector types in https://github.com/g-truc/glm/commit/19a6d64e8c7f65a18235a5aee0112bbe09a6e92f#diff-3b3c579fdecb584260f345edb43e59a7
But somehow quaternions were missed.

**Comment by Groovounet** [comment]
This issue should be fixed in master branch for GLM 0.9.9.1 release.

Thanks,
Christophe

**Comment by martinfalk** [comment]
Thanks!
